src/python/pants/python/python_setup.py

```diff
@@ -74,14 +76,18 @@ def register_options(cls, register):
             "--interpreter-search-paths",
             advanced=True,
             type=list,
-            default=["<PYENV>", "<PATH>"],
+            default=["<ASDF>", "<PYENV>", "<PATH>"],
             metavar="<binary-paths>",
             help=(
                 "A list of paths to search for Python interpreters that match your project's "
                 "interpreter constraints. You can specify absolute paths to interpreter binaries "
                 "and/or to directories containing interpreter binaries. The order of entries does "
                 "not matter. The following special strings are supported:\n\n"
                 '* "<PATH>", the contents of the PATH env var\n'
+                '* "<ASDF>", all python versions currently configured by ASDF '
```

Comment:
Nit, given that this is user-facing documentation (`./pants help`):

```suggestion
                '* "<ASDF>", all Python versions currently configured by ASDF '
```